src/astro/databases/google/bigquery.py

```diff
@@ -135,3 +150,96 @@ def merge_table(
+        location_type = self.OPTIMIZED_PATHS.get(source_file.location.location_type)
+        return bool(location_type and file_type)
+
+    def optimised_transfer(
+        self,
+        source_file: File,
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        **kwargs,
+    ):
+        """
+        Checks if optimised path for transfer between File location to database exists
+        and if it does, it transfers it and returns true else false.
+        """
+        method_name = self.OPTIMIZED_PATHS.get(source_file.location.location_type)
+        if method_name:
+            transfer_method = self.__getattribute__(method_name)
+            transfer_method(
+                source_file=source_file,
+                target_table=target_table,
+                if_exists=if_exists,
+                **kwargs,
+            )
+
+    @staticmethod
+    def get_project_id_from_conn(conn: connection) -> str:
+        """
+        Get project id from conn
+
+        :param conn: Airflow's connection
+        """
+        if conn.extra_dejson and conn.extra_dejson.get("project"):
+            return str(conn.extra_dejson["project"])
+        raise ValueError(f"conn_id {conn.conn_id} has no project id.")
+
+    def gs_to_bigquery(
+        self,
+        source_file: File,
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        **kwargs,
+    ) -> None:
+        """
+        Transfer data from gcs to bigquery
+
+        :param source_file: Source file that is used as source of data
+        :param target_table: Table that will be created on the bigquery
+        :param if_exists: Overwrite table if exists. Default 'replace'
+        """
+        bq_hook = BigQueryHook(
+            gcp_conn_id=target_table.conn_id,
+        )
+
+        write_disposition_val = {"replace": "WRITE_TRUNCATE", "append": "WRITE_APPEND"}
+
+        conn = self.hook.get_connection(target_table.conn_id)
+
+        load_job_config = {
+            "sourceUris": [source_file.path],
+            "destinationTable": {
+                "projectId": self.get_project_id_from_conn(conn),
+                "datasetId": target_table.metadata.schema,
+                "tableId": target_table.name,
+            },
+            "createDisposition": "CREATE_IF_NEEDED",
+            "writeDisposition": write_disposition_val[if_exists],
+            "sourceFormat": self.OPTIMIZED_PATHS_SUPPORTED_FILE_TYPES[
+                source_file.type.name
+            ],
+            "autodetect": True,
+        }
+
+        # Since bigquery has other options besides used here, we need to expose them to end user.
+        # https://cloud.google.com/bigquery/docs/reference/rest/v2/Job#JobConfigurationLoad
+        load_job_config.update(kwargs)
+
+        job_config = {
+            "jobType": "LOAD",
+            "load": load_job_config,
+            "labels": {"target_table": target_table.name},
+        }
+        bq_hook.insert_job(
```

Comment:
Can we not change this to `self.hook.insert_job` and remove `bq_hook`?

Reply:
Done.